scripts/qemu-dm-wrapper

```diff
@@ -13,94 +13,125 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU Lesser General Public License for more details.
 
+import os
 import subprocess
+import sys
 
 path = "/sbin:/usr/sbin:/bin:/usr/bin"
 
 xenstore_write_cmd = "xenstore-write"
+xenstore_read_cmd = "xenstore-read"
 
-# Set cgroup_slice to the name of the cgroup slice the qemu-dm process should live in.
+# Set cgroup_slice to the name of the cgroup slice the qemu-dm process
+# should live in.
 #  - None means leave in the same slice as the parent process.
 #  - '' means move it into the default slice.
 #  - 'system.slice' means move it into the system slice, etc.
-# If the nominated slice does not already exist, the process will be left in its parent's slice.
+# If the nominated slice does not already exist, the process will be
+# left in its parent's slice.
 cgroup_slice = ''
 
 def doexec(args):
-	"""Execute a subprocess, then return its return code, stdout and stderr"""
-	proc = subprocess.Popen([ "/usr/bin/env", "PATH=%s" % path ] + args,stdin=None,stdout=subprocess.PIPE,stderr=subprocess.PIPE,close_fds=True)
-	rc = proc.wait()
-	stdout = proc.stdout
-	stderr = proc.stderr
-	return (rc, stdout, stderr)
-
-def xenstore_write(path, value):
-	(rc, stdout, stderr) = doexec([xenstore_write_cmd, path, value])
+    """Execute a subprocess, then return its return code, stdout and stderr"""
+    proc = subprocess.Popen(["/usr/bin/env", "PATH=%s" % path] + args,
```

Comment:
The environment (including `PATH`) can be passed as a dictionary to `subprocess.Popen` such that using `env` is not necessary. https://docs.python.org/2/library/subprocess.html#popen-constructor